test/test_on_demand_csfle.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2019-present MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Test client side encryption with on demand credentials."""
+import os
+import sys
+import unittest
+
+sys.path[0:0] = [""]
+
+from test import IntegrationTest, client_context
+
+from bson.codec_options import CodecOptions
+from pymongo.encryption import _HAVE_PYMONGOCRYPT, ClientEncryption, EncryptionError
+
+
+class TestonDemandGCPCredentials(IntegrationTest):
+    @classmethod
+    @unittest.skipUnless(_HAVE_PYMONGOCRYPT, "pymongocrypt is not installed")
+    @client_context.require_version_min(4, 2, -1)
+    def setUpClass(cls):
+        super(TestonDemandGCPCredentials, cls).setUpClass()
+
+    def setUp(self):
+        super(TestonDemandGCPCredentials, self).setUp()
+        self.master_key = {
+            "projectId": "devprod-drivers",
+            "location": "global",
+            "keyRing": "key-ring-csfle",
+            "keyName": "key-name-csfle",
+        }
+
+    @unittest.skipIf(not os.getenv("TEST_FLE_GCP_AUTO"), "Not testing FLE GCP auto")
+    def test_01_failure(self):
+        if os.getenv("SUCCESS"):
```

Comment:
`os.getenv("SUCCESS"):` is always True so we'll always skip this test.

Reply:
`SUCCESS=false` in the `testgcpkms-fail-task`

Reply:
Yes but the string "false" is still truthy.

Reply:
Ah, fixed